### PR TITLE
fix(nuxt): display errors in client overlay with `typeCheck`

### LINKED ISSUE #20851: `vite-plugin-checker` not currently working when `typeCheck: true`
issue documented and workaround provided at https://vite-plugin-checker.netlify.app/faq/integration.html#overlay-does-not-display-in-development-mode-1

see https://github.com/nuxt/nuxt/issues/14336#issuecomment-1547588854

### PR DESCRIPTION
**Pull request by aleksdotbar**
### 🔗 Linked issue
resolves #20851

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by xanderbarkhatov** [comment]
What do you think? @danielroe

File: packages/nuxt/src/core/nuxt.ts [hunk 1]
```diff
@@ -308,6 +308,22 @@ async function initNuxt (nuxt: Nuxt) {
     addPlugin(resolve(nuxt.options.appDir, 'plugins/preload.server'))
   }
 
+  // Add type-checking
+  if (process.dev && nuxt.options.builder === '@nuxt/vite-builder' && nuxt.options.typescript.typeCheck === true) {
```
Comment (danielroe) [inline packages/nuxt/src/core/nuxt.ts:312]: I think this section should ideally be in the vite builder package but the plugin approach seems reasonable.

Comment (xanderbarkhatov) [reply packages/nuxt/src/core/nuxt.ts:312]: Yeah, I tried to put it right after `addVitePlugin` part, but couldn't get it to work.
`addTemplate` does nothing at that point.

Comment (danielroe) [reply packages/nuxt/src/core/nuxt.ts:312]: Yes, that does make sense. 👍 I wonder if instead we could add the import to the entry in the vite client config?

Comment (xanderbarkhatov) [reply packages/nuxt/src/core/nuxt.ts]: Updated the code. Hope I understood correctly.

Comment (danielroe) [reply packages/nuxt/src/core/nuxt.ts]: Thanks! Yes, perfect 👍

**Review by danielroe — APPROVED** [review]

**Pull request merged**